integration-tests/src/test/java/org/apache/fineract/integrationtests/DelinquencyBucketsIntegrationTest.java

```diff
@@ -0,0 +1,161 @@
+                responseSpec, jsonRange);
+        jsonRange = DelinquencyRangesHelper.getAsJSON(1, 7);
+        assertNotNull(delinquencyRangeResponse01);
+
+        // when
+        final PutDelinquencyRangeResponse delinquencyRangeResponse02 = DelinquencyRangesHelper.updateDelinquencyRange(requestSpec,
+                responseSpec, delinquencyRangeResponse01.getResourceId(), jsonRange);
+        final GetDelinquencyRangesResponse range = DelinquencyRangesHelper.getDelinquencyRange(requestSpec, responseSpec,
+                delinquencyRangeResponse01.getResourceId());
+        final PutDelinquencyRangeResponse deleteDelinquencyRangeResponse = DelinquencyRangesHelper.deleteDelinquencyRange(requestSpec,
+                responseSpec, delinquencyRangeResponse01.getResourceId());
+
+        // then
+        assertNotNull(delinquencyRangeResponse02);
+        assertNotNull(deleteDelinquencyRangeResponse);
+        assertNotNull(range);
+        assertNotEquals(3, range.getMaximumAgeDays());
+        assertEquals(1, range.getMinimumAgeDays());
+        assertEquals(7, range.getMaximumAgeDays());
+    }
+
+    @Test
+    public void testDelinquencyBuckets() {
+        // given
+        ArrayList<Integer> rangeIds = new ArrayList<>();
+        String jsonRange = DelinquencyRangesHelper.getAsJSON(1, 3);
+        PostDelinquencyRangeResponse delinquencyRangeResponse = DelinquencyRangesHelper.createDelinquencyRange(requestSpec, responseSpec,
+                jsonRange);
+        rangeIds.add(delinquencyRangeResponse.getResourceId());
+        jsonRange = DelinquencyRangesHelper.getAsJSON(4, 30);
+        // Create
+        delinquencyRangeResponse = DelinquencyRangesHelper.createDelinquencyRange(requestSpec, responseSpec, jsonRange);
+        rangeIds.add(delinquencyRangeResponse.getResourceId());
+        String jsonBucket = DelinquencyBucketsHelper.getAsJSON(rangeIds);
+        PostDelinquencyBucketResponse delinquencyBucketResponse = DelinquencyBucketsHelper.createDelinquencyBucket(requestSpec,
+                responseSpec, jsonBucket);
+        // Update
+        jsonRange = DelinquencyRangesHelper.getAsJSON(31, 60);
+        rangeIds.add(delinquencyRangeResponse.getResourceId());
+        jsonBucket = DelinquencyBucketsHelper.getAsJSON(rangeIds);
+        PutDelinquencyBucketResponse updateDelinquencyBucketResponse = DelinquencyBucketsHelper.updateDelinquencyBucket(requestSpec,
+                responseSpec, delinquencyBucketResponse.getResourceId(), jsonBucket);
+        delinquencyRangeResponse = DelinquencyRangesHelper.createDelinquencyRange(requestSpec, responseSpec, jsonRange);
+        rangeIds.add(delinquencyRangeResponse.getResourceId());
+        // Delete
+        PutDelinquencyBucketResponse deleteDelinquencyBucketResponse = DelinquencyBucketsHelper.deleteDelinquencyBucket(requestSpec,
+                responseSpec, delinquencyRangeResponse.getResourceId());
```

Comment:
@josehernandezfintecheandomx I think this is wrong! You cannot use the "delinquencyRangeResponse.getResourceId" to delete DelinquencyBucket...

Reply:
Done!